types/reveal/reveal-tests.ts

```diff
@@ -1,118 +1,809 @@
-Reveal.initialize({
+import Reveal from 'reveal.js';
+import RevealHighlight from 'reveal.js/plugin/highlight/highlight';
+import RevealMarkdown from 'reveal.js/plugin/markdown/markdown';
+import RevealMath from 'reveal.js/plugin/math/math';
+import RevealNotes from 'reveal.js/plugin/notes/notes';
+import RevealSearch from 'reveal.js/plugin/search/search';
+import RevealZoom from 'reveal.js/plugin/zoom/zoom';
+
+// define useful constants
+const el = document.querySelector<HTMLElement>('.deck')!;
+
+// --------------- //
+// Reveal instance //
+// --------------- //
+
+// use two arguments
+// $ExpectType Api
+new Reveal(el, {
+    plugins: [RevealMarkdown, RevealHighlight, RevealNotes, RevealSearch, RevealMath, RevealZoom],
+});
+
+// use one arguments
+// $ExpectType Api
+new Reveal({
+    plugins: [RevealMarkdown, RevealHighlight, RevealNotes, RevealSearch, RevealMath, RevealZoom],
+});
+
+// without argument
+// $ExpectType Api
+new Reveal();
+
+// $ExpectType Promise<Api>
+Reveal.initialize();
+
+// ----------------- //
+// initialize Method //
+// ----------------- //
+
+const deck = new Reveal();
+
+// $ExpectType Promise<Api>
+deck.initialize({
+    // default values from https://github.com/hakimel/reveal.js/blob/master/js/config.js
+
+    // The "normal" size of the presentation, aspect ratio will be preserved
+    // when the presentation is scaled to fit different resolutions
+    width: 960,
+    height: 700,
+
+    // Factor of the display size that should remain empty around the content
+    margin: 0.04,
+
+    // Bounds for smallest/largest possible scale to apply to content
+    minScale: 0.2,
+    maxScale: 2.0,
+
+    // Display presentation control arrows
     controls: true,
+
+    // Help the user learn the controls by providing hints, for example by
+    // bouncing the down arrow when they first encounter a vertical slide
+    controlsTutorial: true,
+
+    // Determines where controls appear, "edges" or "bottom-right"
+    controlsLayout: 'bottom-right',
+
+    // Visibility rule for backwards navigation arrows; "faded", "hidden"
+    // or "visible"
+    controlsBackArrows: 'faded',
+
+    // Display a presentation progress bar
     progress: true,
-    history: true,
+
+    // Display the page number of the current slide
+    // - true:    Show slide number
+    // - false:   Hide slide number
+    //
+    // Can optionally be set as a string that specifies the number formatting:
+    // - "h.v":	  Horizontal . vertical slide number (default)
```

Comment:
<!--
  1 failure:  The root .editorc...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: types/reveal/reveal-tests.ts;
  Line: 79;
-->
- :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.